jfix-zookeeper/src/main/java/ru/fix/zookeeper/lock/PersistentExpiringLockManager.java

```diff
@@ -180,4 +153,38 @@ public void close() {
         locks.clear();
         lockProlongationScheduler.close();
     }
+
+    /**
+     * Try to prolong lock or else remove it from {@link #locks}
+     */
+    private void prolongOrRemoveLock(LockIdentity lockId, LockContainer lockContainer) {
+        boolean prolonged = false;
+        try {
+            logger.debug("Check and prolong lockId={}", lockId);
+            prolonged = lockContainer.lock.checkAndProlongIfExpiresIn(
+                    config.get().getLockAcquirePeriod(),
+                    config.get().getExpirationPeriod()
+            );
+        } catch (Exception e) {
+            // Lock could be already removed and closed with {@link #release(LockIdentity)} in another thread
+            if (!locks.containsKey(lockId)) {
+                logger.info(
+                        "Failed to checkAndProlongIfExpiresIn lockId {} cause it has been deleted already", lockId, e
```

Comment:
If we invoke `PersistentExpiringDistributedLock#checkAndProlong` on the lock which underlying node was removed from ZK we will get `false` result, not an Exception. We can get an Exception in case of network problems. 
In current case we are  dumping this exception in INFO log, which is not correct. 
Suggestion: remove locks.containsKey and logger.info block from exception handling section

Suggestion: 
```
boolean prolonged = false;
try {
    logger.debug("Check and prolong lockId={}", lockId);
    prolonged = lockContainer.lock.checkAndProlongIfExpiresIn(
            config.get().getLockAcquirePeriod(),
            config.get().getExpirationPeriod()
    );
} catch (Exception e) {
        //do not care if lock already removed from locks or not, exception indicate technical problem like network
        logger.error("Failed to checkAndProlongIfExpiresIn persistent locks with lockId {}", lockId, e);
}
if (!prolonged) {
    if (!locks.containsKey(lockId)) {
           // Lock could be already removed and closed with {@link #release(LockIdentity)} in another thread
          // Normal case, do  not pollute logs
          return;
   }

    locks.remove(lockId);
    logger.error("Failed lock prolongation for lock={}. Lock is removed from manager", lockId);
    try {
        lockContainer.prolongationFailedListener.onLockProlongationFailedAndRemoved(lockId);
    } catch (Exception exc) {
        logger.error("Failed to invoke ProlongationFailedListener on lock {}", lockId, exc);
    }
}
```

Reply:
mb decrease level to warning in the catch log? Just because we will get the error  log next

Reply:
>Just because we will get the error log next

Did not get it, could you please elaborate on that?

Reply:
I mean that If `lock.checkAndProlongIfExpiresIn` returns false or thrown exception there will be logged error `Failed lock prolongation for lock...`.